api/serializers.py

```diff
@@ -19,21 +21,40 @@ class Meta:
             'cohort': {'min_value': 0}
         }
 
+    def get_full_name(self, obj):
+        return "{} {}".format(
+            obj.first_name,
+            obj.last_name
+        )
+
     def create(self, validated_data):
-        password = validated_data.pop('password')
         user = User(**validated_data)
-        user.set_password(password)
         user.save()
         return user
 
 
-class ItemSerializer(serializers.ModelSerializer):
+class AssetSerializer(serializers.ModelSerializer):
+    checkin_status = serializers.SerializerMethodField()
+
     class Meta:
         model = Item
         fields = ("id", "item_code", "serial_number", "model_number",
-                  "status", "assigned_to", "created_at", "last_modified",
+                  "allocation_status", "checkin_status", "assigned_to",
+                  "created_at", "last_modified",
                   )
 
+    def get_checkin_status(self, obj):
+        try:
+            asset_log = AssetLog.objects.filter(asset=obj) \
+                .order_by('-created_at').first()
+
+            if asset_log.log_type == AssetLog.CHECKIN:
+                return "checked_in"
+            elif asset_log.log_type == AssetLog.CHECKOUT:
```

Comment:
we should also take care of when no logs have been entered for the asset yet.. eg for a newly added asset